### PR TITLE
Added csv output format

### PR DESCRIPTION
**Pull request by edsonarios**
Issue: #1320
```
softlayer-python/ () $ slcli --format csv vs list
id,hostname,domain,deviceStatus.name,datacenter,primary_ip,backend_ip,createDate,action
1002,adns,vmware.chechu.com,Running,dal10,48.110,122.174,2020-04-06T05:28:15-06:00,-
1323,easerverh,easerverd.cloud,Running,dal13,227.82,211.196,2022-08-11T10:52:46-06:00,-
1204,fotest,test.local,Running,dal05,148.28,12.198,2021-05-19T12:29:55-06:00,-
1278,lab-web,test.com,Running,dal05,148.27,12.203,2022-02-08T08:31:46-06:00,-
1278,lab-web,test.com,Running,dal05,148.26,12.220,2022-02-08T08:33:12-06:00,-
1023,reserved,example.com,Stopped,dal13,-,-,2020-05-12T14:10:22-06:00,-
1096,techsupport,SoftLayer-Internal-Development-Community.cloud,Running,sao01,169.57.227.67,10.151.109.245,2020-09-25T09:34:12-06:00,-
```
Note: Possible upgrade options
Add option to download the output as a file .csv
Add option to change the deliminator, now by default ','

### CODE REVIEW

File: SoftLayer/CLI/formatting.py [hunk 1]
```diff
@@ -440,3 +444,29 @@ def _format_list_objects(result):
...
     return table
+
+
+def csv_output_format(data, delimiter=','):
+    """Formating a table to csv format and show it."""
+    data = clean_null_table_rows(data)
+    with tempfile.TemporaryDirectory() as temp_file:
+        f_name = os.path.join(temp_file, 'temp_csv_file')
+        with open(f_name, 'w', encoding='UTF8') as file:
+            writer = csv.writer(file, delimiter=delimiter)
```
Comment (allmightyspiff) [inline SoftLayer/CLI/formatting.py:455]: I think you can just pass `sys.stdout` as the filename to print the result to the console. Don't create a file just to read from it again.
```suggestion
            writer = csv.writer(sys.stdout, delimiter=delimiter)
```

**Review by allmightyspiff — CHANGES_REQUESTED** [review]
- Use `sys.stdout` to output csv directly instead of to a file which you then read from.
- Add some unit tests for this.
- Test  a nested table to see how that looks as well.

**Review by BrianSantivanez — COMMENTED** [review]
Running the changes 
image
image
After the changes looks good to me

File: SoftLayer/CLI/formatting.py (hunk 1 above)
Comment (edsonarios) [reply SoftLayer/CLI/formatting.py]: Used sys.stdout for print csv datas
Added some unit test
Supported nested tables to csv outputformat

**Comment by edsonarios** [comment]
Tables
image
Nested tables
image

**Review by allmightyspiff — APPROVED** [review]

**Pull request merged**